[Pytorch] add an option to disable TORCH_WARN and TORCH_WARN_ONCE log

biubiuX opened the pull request:
Summary: Add an option to disable TORCH_WARN, some op could trigger spammy TOCH_WARN log which is not desired under certain scenario.

Test Plan:
Tested with
-pt.disable_warn = 1 and -pt.disable_warn = 0

verified TORCH_WARN and TORCH_WARN_ONCE are properly handled

tested with
-pt.strip_error_messages = 1, -pt.disable_warn = 0

verified strip error message is respected when warn is printed

Differential Revision: D40321550

kurtamohler commented:
Looks good to me, but there's a recent merge conflict

biubiuX commented:
@kurtamohler mind taking another look ? thanks !

kurtamohler commented:
Looks good, thanks!

Just to make sure you're aware, you can filter all warnings at the Python level like this:

```python
import warnings
warnings.filterwarnings('ignore')
```

But this also filters out all warnings in your project, not just the PyTorch ones, so I can see why this `DISABLE_WARN` flag is useful.

There is a loose plan to add a warning filter to the C++ side of PyTorch (part of https://github.com/pytorch/rfcs/pull/43). The details aren't decided yet, but we could potentially add a Python binding for it. If/when that exists, you could use that instead of having to build PyTorch yourself

biubiuX commented:
Thanks for the tip. We're the consumer of the model and we don't directly work on the python code. We're working with our partner to fix/adjust warning, but it would be great if we also have log/warning config on c++ side.

biubiuX commented:
not sure if the workflow failure is caused by my change, is there anything I can do here ?

kurtamohler commented:
Yeah these failures don't seem to be related. Try rebasing just to make sure

biubiuX commented:
Could anyone help me run the workflows ?

kurtamohler commented:
The two failing jobs actually do seem to be caused by your changes. I think I may have steered you wrong earlier--I had only seen the `Device or resource busy` error in the logs and assumed it wasn't related. But there's another error that happens earlier in the logs:
```
C:\actions-runner\_work\pytorch\pytorch\c10/core/TensorImpl.h(1758): error C2275: 'c10::UserWarning': illegal use of this type as an expression
C:\actions-runner\_work\pytorch\pytorch\c10/core/TensorImpl.h(1758): note: see declaration of 'c10::UserWarning'
C:\actions-runner\_work\pytorch\pytorch\c10/core/TensorImpl.h(1758): error C2064: term does not evaluate to a function taking 0 arguments
```

biubiuX committed:
[Pytorch] add an option to disable TORCH_WARN and TORCH_WARN_ONCE log (#87188)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/87188

Add an option to disable TORCH_WARN, some op could trigger spammy TOCH_WARN log which is not desired under certain scenario.

Test Plan:
Tested with
-pt.disable_warn = 1 and -pt.disable_warn = 0

verified TORCH_WARN and TORCH_WARN_ONCE are properly handled

tested with
-pt.strip_error_messages = 1, -pt.disable_warn = 0

verified strip error message is respected when warn is printed

Differential Revision: D40321550

fbshipit-source-id: 74e0997f7271336543bad2691e327006d6cda7f9

biubiuX commented:
@kurtamohler mind taking another look(also trigger the workflow) ? It indeed seems to be a subtle preprocessor bug in MVSC(gcc, clang works ok).

biubiuX commented:
gentle ping @kurtamohler ; )

kurtamohler commented:
@pytorchbot merge

kurtamohler commented:
Oh right, forgot we need @ezyang's approval before merging

biubiuX commented:
@pytorchbot merge
